system/jlib/jden.hpp

```diff
@@ -0,0 +1,37 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef JDEN_INCL
+#define JDEN_INCL
+
+#include "jlzw.hpp"
+
+#define DENCOMPRESSEDFILEBLOCKSIZE (0x100000)
+
+extern jlib_decl ICompressor *createDENCompressor();
+extern jlib_decl IExpander   *createDENExpander();
+
+extern jlib_decl void DENCompressToBuffer(MemoryBuffer & out, size32_t len, const void * src);
+extern jlib_decl void DENDecompressToBuffer(MemoryBuffer & out, const void * src);
+extern jlib_decl void DENDecompressToBuffer(MemoryBuffer & out, MemoryBuffer & in);
+extern jlib_decl void DENDecompressToAttr(MemoryAttr & out, const void * src);
+extern jlib_decl void DENDecompressToBuffer(MemoryAttr & out, MemoryBuffer & in);
```

Comment:
It's a bit of a pity there isn't a generic:
```
compressToBuffer(ICompressor *comp, size32_t len, const void *src);
```

For e.g. fastlz and den it would less efficient as it stands, as the implementation has to build up the input stream and then write it out to the output buffer.
But we could introduce something like : 
```
ICompressor::compress(size32_t len, const void *src, MemoryBuffer &out);
```
..to be used for single shot compressions like these methods.
Then users of XXXcompressToBuffer methods could instead use an existing ICompressor/IExpander and make the use of these calls configurable.